index.html

```diff
@@ -945,292 +651,34 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
-Once connection is established set the value of <a>nextHopProtocol</a>
-to the ALPN ID used by the connection. If a connection can not be
-established, record the time up to the connection failure as
-<a>connectEnd</a> and go to the step labeled <a
-href= "#dfn-step-final-record">final record</a>.</li>
-<li>The user agent MUST set the <a>secureConnectionStart</a>
-attribute as follows:
-<ol>
-<li>When a secure transport is used, the user agent MUST record the
-time as <a>secureConnectionStart</a> immediately before the
-handshake process to secure the connection.</li>
-<li>When a secure transport is not used, the user agent MUST set
-the value of <a>secureConnectionStart</a> to 0.</li>
-</ol>
-</li>
-<li><em>Request start:</em> <dfn
-data-lt="step-request-start">Immediately before a user agent starts
-sending the request for the resource</dfn>, record the
-<a>current time</a> as <a>requestStart</a>. If a user agent needed
-multiple retries to send the request, record the <a>current time</a>
-of the last attempt.
-<p class=note>Network protocols may not perform the connection
-establishment, secure connection establishment and request sending in a
-sequential manner. Therefore, developers should not expect these values
-to always be in a particular order. </p>
-</li>
-<li>Record the time as <a>responseStart</a> <dfn data-lt=
-"step-response-start">immediately after the user agent receives the
-first byte of the response</dfn>.</li>
-<li><em>Response end:</em> Record the time as <a>responseEnd</a> <dfn
-data-lt="step-response-end">immediately after receiving the last byte of
-the response</dfn>.
-<ol>
-<li>Return to the step labeled <a href="#dfn-step-connect-start">connect start</a> if the
-user agent fails to send the request or receive the entire
-response, and needs to reopen the connection.
-<aside class="example">
-<p>When <a data-cite="RFC7230#section-6.3">persistent
-connection</a> [[RFC7230]] is enabled, a user agent may first try
-to re-use an open connect to send the request while the connection
-can be <a data-cite="RFC7230#section-6.5">asynchronously
-closed</a>. In such case, <a>connectStart</a>, <a>connectEnd</a>
-and <a>requestStart</a> SHOULD represent timing information
-collected over the re-open connection.</p>
-</aside>
-</li>
-<li>Set the value of <a>transferSize</a>, <a>encodedBodySize</a>,
-<a>decodedBodySize</a> to corresponding values, subject to
-<a>timing allow check</a> algorithm.</li>
-</ol>
-</li>
-<li><em>Final record:</em> If <a
-data-link-for="PerformanceResourceTiming">responseEnd</a>
-is not set, set it to the current time. <dfn data-lt=
-"step-final-record">Record the difference between
-<a>responseEnd</a> and <a>startTime</a> in
-<a>duration</a></dfn>.</li>
-<li>If the fetched resource results in an HTTP redirect or
-<a data-cite="HTML#or-equivalent" data-lt=
-'HTTP response codes equivalence'>equivalent</a>, then
+<li><a>Queue a global task</a> on the <a>networking task source</a> with
```

Comment:
Why do we need a task here? Aren't we already on a task so this can be done synchronously?

Reply:
Not necessarily. Someone needs to queue a task on the global object. We can have FETCH queue this task on the `taskDestination` instead, but there is no guarantee that it's the global object.
Since we anyway need a global object for the time origin, I thought making this re-entrant and queue it here would be right, but open to do this in a different way.